scripts/predict_trader/report.py

```diff
@@ -333,12 +333,34 @@ def _parse_args() -> Any:
                 mech_contract_address = activity_checker_contract.functions.agentMech().call(block_identifier=current_block_number)
 
             mech_contract_data = requests.get(MECH_CONTRACT_JSON_URL).json()
-
             mech_contract_abi = mech_contract_data.get("abi", [])
-
+            mech_contract_abi.append({
+                "inputs": [
+                    {
+                    "internalType": "address",
+                    "name": "account",
+                    "type": "address"
+                    }
+                ],
+                "name": "getRequestCount",
+                "outputs": [
+                    {
+                    "internalType": "uint256",
+                    "name": "",
+                    "type": "uint256"
+                    }
+                ],
+                "stateMutability": "view",
+                "type": "function"
+            })
             mech_contract = w3.eth.contract(
                 address=mech_contract_address, abi=mech_contract_abi   # type: ignore
             )
+            try:
+                mech_request_count = mech_contract.functions.mapRequestsCounts(safe_address).call(block_identifier=current_block_number)
+            except (ABIFunctionNotFound, ValueError):
+                # Use mapRequestCounts for newer mechs
+                mech_request_count = mech_contract.functions.mapRequestCounts(safe_address).call(block_identifier=current_block_number)
```

Comment:
Is the `use_mm` flag available here? Maybe you could avoid this by checking it.

Reply:
@Adamantios it can be taken from configured environment variables. But the local config can differ from the on-chain state, if the user manually tweaks something or abruptly stops the script.